### DISCUSSION TITLE
Sprite: Memory Crash

### DISCUSSION THREAD

**Discussion by audreylim** [Bodmer/TFT_eSPI#1663] [Q&A - Sprites] [answered]
Hello, I'm trying to draw a sprite on my LCD. I'm also doing a bunch of other API calls with the ESP8266. I've some certs in SPIFFS which is loaded with

```
  wiFiClient.loadCertificate(cert)
```

When combined with the sprite drawing, adding this line crashes the program: https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Sprite/Sprite_scroll/Sprite_scroll.ino#L45

```
--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (29):
epc1=0x4000df64 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000

.....
 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 3584, room 16 
tail 0
chksum 0xb0
csum 0xb0
v2843a5ac
~ld
```

Would you know of any way to optimize memory usage to prevent the crash?

**Comment by Bodmer** [comment, marked as answer]
I suspect you are running out of RAM, WiFi functions can consume a lot. When a sprite is created it requests RAM, in this case about 8kbytes will be requested. Subsequently the Wifi stack demands more Ram and the processor crashes. You could add code to print the heap size in various places to see how much RAM is available. As your project evolves it is likely to grow and demand more RAM, so upgrading to an ESP32 is worthwhile.

**Reply by audreylim** [reply]
Hmm, I think you're right. Thank you!
